### PR TITLE
added new config to allow kafka record timestamp to set as published_…

### PR DESCRIPTION
**Pull request by nyk0322**
…date event attribute

### What does this PR do?

A brief description of the change being made with this pull request.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

File: src/main/java/com/datadoghq/connect/logs/sink/DatadogLogsSinkConnectorConfig.java [hunk 1]
```diff
@@ -33,6 +33,7 @@ public class DatadogLogsSinkConnectorConfig extends AbstractConfig {
     private static final String DD_URL_FORMAT_FROM_SITE = "http-intake.logs.%s:443";
     private static final String DEFAULT_DD_SITE = "datadoghq.com";
     public static final String DEFAULT_DD_URL = String.format(DD_URL_FORMAT_FROM_SITE, DEFAULT_DD_SITE);
+    public static final String USE_RECORD_TIMESTAMP = "datadog.use.record.timestamp";
```
Comment (jszwedko) [inline src/main/java/com/datadoghq/connect/logs/sink/DatadogLogsSinkConnectorConfig.java:36]: Could we call this `datadog.add_published_date`? I think that's a bit clearer. I'm not sure we'll end up with additional options in a `datadog.use.record` namespace where it'd be make sense to have an option namespaced like that.

File: src/test/java/com/datadoghq/connect/logs/sink/DatadogLogsApiWriterTest.java [hunk 2]
```diff
@@ -153,4 +154,24 @@ public void metadata_asOneBatch_shouldPopulatePerBatch() throws IOException {
 
         Assert.assertEquals("[{\"message\":\"someValue1\",\"ddsource\":\"kafka-connect\",\"ddtags\":\"topic:someTopic,team:agent-core,author:berzan\",\"hostname\":\"test-host\",\"service\":\"test-service\"},{\"message\":\"someValue2\",\"ddsource\":\"kafka-connect\",\"ddtags\":\"topic:someTopic,team:agent-core,author:berzan\",\"hostname\":\"test-host\",\"service\":\"test-service\"}]", request.getBody());
     }
+
+    @Test
+    public void writer_withUseRecordTimeStampEnabled_shouldPopulateRecordTimestamp() throws IOException {
```
Comment (jszwedko) [inline src/test/java/com/datadoghq/connect/logs/sink/DatadogLogsApiWriterTest.java:159]: Thanks for adding a test!

File: src/main/java/com/datadoghq/connect/logs/sink/DatadogLogsSinkConnectorConfig.java [hunk 3]
```diff
@@ -166,7 +169,13 @@ private static void addMetadataConfigs(ConfigDef configDef) {
                 ++orderInGroup,
                 Width.LONG,
                 "Datadog logs site"
-        );
+        ).define(
```
Comment (jszwedko) [inline src/main/java/com/datadoghq/connect/logs/sink/DatadogLogsSinkConnectorConfig.java:172]: I think we'll just need to add this to the README as well to document it.

File: src/main/java/com/datadoghq/connect/logs/sink/DatadogLogsSinkConnectorConfig.java [hunk 4]
```diff
@@ -166,7 +169,13 @@ private static void addMetadataConfigs(ConfigDef configDef) {
...
+                USE_RECORD_TIMESTAMP,
+                Type.BOOLEAN,
+                false,
+                null,
+                Importance.MEDIUM,
+                "Valid settings are true or false. When set to `true`, The timestamp is retrieved from the Kafka record and passed to Datadog");
```
Comment (jszwedko) [inline src/main/java/com/datadoghq/connect/logs/sink/DatadogLogsSinkConnectorConfig.java:178]:
```suggestion
                "Valid settings are true or false. When set to `true`, The timestamp is retrieved from the Kafka record and passed to Datadog as `published_date`");
```

**Review by jszwedko — COMMENTED** [review]
Thanks for this @nyk0322 ! I left a few comments but this seems like a good addition to me. I might even go so far as to say it should be the default behavior to add the timestamp, but we can start with opt-in for now to preserve the existing behavior.

I tested it out locally and observed the records ending up in Datadog with the `published_date` added.

File: src/main/java/com/datadoghq/connect/logs/sink/DatadogLogsSinkConnectorConfig.java (hunk 1 above)
Comment (nyk0322) [reply src/main/java/com/datadoghq/connect/logs/sink/DatadogLogsSinkConnectorConfig.java]: changed the parameter name to datadog.add_published_date

File: src/main/java/com/datadoghq/connect/logs/sink/DatadogLogsSinkConnectorConfig.java (hunk 3 above)
Comment (nyk0322) [reply src/main/java/com/datadoghq/connect/logs/sink/DatadogLogsSinkConnectorConfig.java:172]: Added to README file

**Review by jszwedko — APPROVED** [review]
Thanks @nyk0322 ! This looks good to me now. We'll just need docs approval before merging.

**Review by estherk15 — APPROVED** [review]
No docs review needed

**Pull request merged**